src/audio/module_adapter/CMakeLists.txt

```diff
@@ -1,10 +1,11 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
-if(NOT CONFIG_LIBRARY)
-	if (CONFIG_IPC_MAJOR_4 OR CONFIG_CADENCE_CODEC OR (NOT CONFIG_COMP_LEGACY_INTERFACE) OR CONFIG_COMP_FIR)
+
+if (CONFIG_IPC_MAJOR_4 OR CONFIG_CADENCE_CODEC OR (NOT CONFIG_COMP_LEGACY_INTERFACE) OR CONFIG_COMP_FIR)
 	add_local_sources(sof module_adapter.c module/generic.c)
-	endif()
+endif()
 
+if((NOT CONFIG_LIBRARY) OR CONFIG_LIBRARY_STATIC)
```

Comment:
Here again, so I guess this is on purpose...?

Reply:
yep, we need to build in - however this does need to be updated to more meaningful CONFIG names as part of testbench/plugin work.